### PR TITLE
Import/Export: Merge command and performance optimizations

### PR DESCRIPTION
**Pull request by kewitz**
Adding a few features more to the import/export script:
- [x] Dump and restore SequelizeMeta table
- [x] Adds SequelizeMeta table hash to metadata
- [x] Adds merge command to merge data into existing DB
  - [ ] Add missing instructions on how to deduplicate certain models
- [x] Optimize query by combining dependencies queries
- [x] Optimize pagination limit and fix pagination limit vs. recipe limit mixup 
- [x] Add working OSC host-wide recipe
- [x] Fix bug that would prevent us traverse previously parsed entities

### CODE REVIEW

File: server/lib/import-export/sanitize.ts [hunk 1]
```diff
@@ -257,14 +257,15 @@ const PROD_SANITIZERS: { [k in ModelNames]: Sanitizer<k> } = {
...
       };
     }
   },
   PayoutMethod: (payoutMethod, req) => {
     if (!getContextPermission(req as express.Request, PERMISSION_TYPE.SEE_PAYOUT_METHOD_DETAILS, payoutMethod.id)) {
       return {
-        data: null,
-        isSaved: null,
+        data: {},
+        isSaved: false,
```
[the remark this answers is not in the document]
Comment (kewitz) [reply server/lib/import-export/sanitize.ts:268]: The issue here is that isSave is a non-null field. Exporting this as null will cause issues when restoring.
There's a conflict here between exporting valid data or adding the missing values when importing.

File: scripts/smart-dump.ts [hunk 2]
```diff
@@ -155,15 +179,20 @@ program.command('restore <file>').action(async file => {
     logger.info('>>> Inserting Data...');
 
     for await (const line of rl) {
-      const row = JSON.parse(line);
+      let row = JSON.parse(line);
+      if (DEFAULT_VALUES[row.model]) {
+        row = mapValues(row, (value, key) =>
+          !isUndefined(DEFAULT_VALUES[row.model][key]) && isNull(value) ? DEFAULT_VALUES[row.model][key] : value,
+        );
+      }
```
[the remark this answers is not in the document]
Comment (kewitz) [reply scripts/smart-dump.ts:187]: The issue here is that we have non-null properties that have no default value.

File: server/lib/import-export/export.ts [hunk 3]
```diff
@@ -73,48 +73,83 @@ const serialize = async (model: ModelNames, req: PartialRequest, document: Insta
...
   } while (offset < totalCount);
 }
 
 const hashObject = (obj: Record<string, any>) => crypto.hash(JSON.stringify(obj));
 
+const isTargetWhere = q => {
+  const ks = keys(q.where);
+  const vs = values(q.where);
+  return ks.length === 1 && typeof vs[0] !== 'object';
+};
```
[the remark this answers is not in the document]
Comment (kewitz) [reply server/lib/import-export/export.ts:98]: Nice catch!

File: scripts/smart-dump.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Betree) [reply scripts/smart-dump.ts]: Can we fix that directly on the model? In the case of `isSaved`, it has a `defaultValue`: https://github.com/opencollective/opencollective-api/blob/db7e505ab91f8e9c3768862f4d5fba6a194366bf/server/models/PayoutMethod.ts#L288

**Pull request merged**